Developerneva

developernevid opened the pull request:
Added peertopeer empty file;
Fixed typo in authorization.qml;
Replaced header guards to pragma once.

developernevid committed:
[ALL] Fix typos and add syntax replacements

1) Renamed authorization file;
2) Replaced header guards to pragma once

developernevid committed:
[Server] header guard replacement

Changed header guard to pragma once

developernevid committed:
[Client][PeerToPeer][Ignore] Add peertopeer functionality

Add peertopeer functionality with stub.

nefelim commented on src/Client/CMakeLists.txt:
```diff
@@ -60,9 +60,9 @@ include_directories("${CMAKE_CURRENT_SOURCE_DIR}/Headers")
 
 set(CPPSOURCES "Sources/main.cpp" "Sources/client.cpp" "Sources/qmlclientstate.cpp"
-    "Sources/qmlclientsonlinemodel.cpp" "Sources/utils.cpp")
+    "Sources/qmlclientsonlinemodel.cpp" "Sources/utils.cpp" "Sources/peertopeer.cpp")
```
Лучше писать каждый файл в отдельной строчке. При ревью разницы, так будет виднее, что добавилось

nefelim commented on src/Client/Headers/client.h:
```diff
@@ -1,5 +1,4 @@
-#define CLIENT_H
+#pragma once
 #include <grpcpp/grpcpp.h>
 #include <iostream>
```
Заголовочные файлы лучше сортировать так:
1. stdafx.h (precompiled headers только в cpp!)
1. системные/std/boost/qt
2. сторонние библиотеки
3. Свои файлы

nefelim commented on src/Client/Sources/peertopeer.cpp:
```diff
@@ -0,0 +1,6 @@
+#include "peertopeer.h"
+
+PeerToPeer::PeerToPeer()
```
Пустые конструкторы обычно не нужны, иначе добавь в заготовку хотя-бы параметры, которые хочешь пересылать

developernevid replied:
peertopeer это для следующего коммита

nefelim commented on src/Common/client-server.proto:
```diff
@@ -24,3 +28,12 @@ message AuthorizationResponse {
+
+message CallResponse {
+  bool isSuccessful = 1;
+  int32 audiobytes = 2;
```
1. это число может быть меньше 0?
2. Это важный файл. Лучше добавить комментариев, что значит каждое поле
3. Лучше придерживаться одного стиля. Не может быть user*_*names и caller*N*ame  audio*b*ytes. поменяйте плиз на один стиль

developernevid replied:
1. это не целое число, это просто 8 байтов.  Я вот не уверен, какой тип данных лучше подходит - целый int/uint или литерный -string
2. OK
3. OK

nefelim commented on src/Client/Headers/utils.h:
```diff
@@ -10,5 +9,3 @@ namespace secure_voice_call {
     void changeHost(QString &ip, const QString &host);
     void changePort(QString &ip, const unsigned int port);
 }
-
```
const unsigned int port - для маленьких типов по значению можно без const

nefelim reviewed (changes requested):
посмотрите и отформатируйте плиз файлы по такому стилю:
```
namespace A
{
    class B
    {
    public:
        void f(int a)
        {
            if (a != 0)
            {
                m_a = a;
                f(a);
            }
        }
    private:
        int m_a = 0;
}
```

developernevid committed:
[Client][PeerToPeer][Maxim Relay] Syntax and PeerToPeer fixes

Added binding call button to PeerToPeer session.
Corrected syntax misses.

developernevid committed:
[Client][UI]  Added ui components, deleted states

1) Added components for conversation, incoming call, outgoing call
states;
2) Changed qml items directories;
3) Changed principle of displaying ui components - now they are
loaded dynamically.

developernevid committed:
[Client][UI] Removed extra connections, updated ui

1) Replaced connections by Q_INVOKABLE methods
2) Updated out/in-coming call and conversation items

Commit:
Fixed crash with server connection.

developernevid committed:
[Client] Fixed extra output

Fixed extra output and removed using namespace std row.

Commit:
[ALL] Corrections in cmake and added command parser

1) Added to cmake script which automatically  generate runs conan install;
2) Replaced utils class to Tools library
3) Added to Tools library command args parcer
4) All projects now displayed as subdirectories

Commit:
[PeerToPeer] Implemented Accept/Decline methods

Implemented Accept/Decline methods for incoming, outgoing and inConversation qml items.
